lib/ddtrace/contrib/action_cable/events/transmit.rb

```diff
@@ -0,0 +1,48 @@
+require 'ddtrace/contrib/analytics'
+require 'ddtrace/contrib/action_cable/event'
+
+module Datadog
+  module Contrib
+    module ActionCable
+      module Events
+        # Defines instrumentation for 'transmit.action_cable' event
+        module Transmit
+          include ActionCable::Event
+
+          EVENT_NAME = 'transmit.action_cable'.freeze
+
+          module_function
+
+          def event_name
+            self::EVENT_NAME
+          end
+
+          def span_name
+            Ext::SPAN_ACTION
```

Comment:
Are both `PerformAction` and `Transmit` supposed to be `action_cable.action` spans? Why not differentiate with different span names?

Reply:
Both `PerformAction` and `Transmit` are new root spans.
They happen when:
`PerformAction`: the websocket worker thread processes a client request.
`Transmit`: the websocket worker thread broadcasts a message to all connected clients.

My intention was to have all asynchronous jobs executed by `action_cable` to be grouped under the `#{service-name}-action_cable` service name.

The reason that they share the same span name is due to the way our UI selects the most frequent span name to display by default. When I had them separate at first, I was only able to find one of them under the service `#{service-name}-action_cable`.

One alternative I see is creating two distinct service names.

Reply:
I see; makes sense regarding span name.

If these are root spans, how does distributed tracing come into play? Are we saying that since this worker thread runs independently of other requests (like the typical port 80 --> Rack requests), that it should be considered its own trace?

I guess it also raises the question of "can/how should distributed tracing work across these sockets?" I'm interested in the case where we have two Rails services talking to each other via ActionCable.